src/App/App.jsx

```diff
@@ -1,21 +1,69 @@
-import { Switch, Route } from 'react-router-dom';
-import { Home } from '../components/Home/Home';
+import React from 'react';
+import { Switch, Route, Redirect } from 'react-router-dom';
+import { Home } from '../components/Pages/Home/Home';
 import { Members } from '../components/Pages/Members/Members';
 import { Tasks } from '../components/Pages/Tasks/Tasks';
+import { AllTasks } from '../components/Pages/Tasks/AllTasks';
 import { Progress } from '../components/Pages/Progress/Progress';
-import style from './App.module.css';
+import { Header } from '../components/Common/Header/Header';
+import { Footer } from '../components/Common/Footer/Footer';
+import { Login } from '../components/Pages/Login/Login';
+import { Track } from '../components/Pages/Track/Track';
 
-export const App = () => {
-  return (
-    <div className={style.content}>
-      <Switch>
-        <Route exact path='/'>
-          <Home />
-        </Route>
-        <Route path='/members' component={Members} />
-        <Route path='/progress/:id' component={Progress} />
-        <Route path='/tasks/:id' component={Tasks} />
-      </Switch>
-    </div>
-  );
-};
+export class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isAuth: false,
+      userEmail: '',
+    };
+  }
+
+  setAuth = (userEmail) => {
+    if (userEmail) {
+      this.setState({ userEmail, isAuth: true });
+    }
+  };
+
+  logout = () => {
+    this.setState({ userEmail: '', isAuth: false });
+  };
```

Comment:
create common `Provider` component for authentication by using https://ru.reactjs.org/docs/context.html#reactcreatecontext

Reply:
```
  logoutHandler = () => {
    this.setState(initialStateAuth);
  };
```

Reply:
Created `AuthProvider`.

Reply:
Created `AuthProvider`. 
```suggestion
export class AuthProvider extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      ...initialStateAuth,
      loginHandler: this.login,
      logoutHandler: this.logout,
    };
  }

  setAuth = (name, role, uid) => {
    this.setState((prevState) => ({ ...prevState, name, role, uid, isAuth: true }));
  };

  logout = () => {
    this.setState((prevState) => ({ ...prevState, ...initialStateAuth }));
  };

  login = async (email, password) => {
    this.resetError();
    try {
      const { role, name, uid } = await findUser(email, password);
      if (uid) {
        this.setAuth(name, role, uid);
      } else {
        this.setState({ error: 'user not found' });
      }
    } catch (err) {
      console.log(err);
      this.setState({ error: 'user not found' });
    }
  };

  resetError() {
    this.setState({ error: '' });
  }

  render() {
    const { children } = this.props;

    return <AuthContext.Provider value={this.state}>{children}</AuthContext.Provider>;
  }
}

AuthProvider.propTypes = {
  children: PropTypes.node.isRequired,
};
```

Reply:
That's strange that you do `resetError` every time at the begging, you should do it in the end and only when you have error. If you log error for some reason in console you should use `console.error`. I've already told it 3 times 😄

Reply:
Deleted resetError() from login in Provider and call from form handler.